src/MDMTransaction.m

```diff
@@ -30,10 +32,25 @@ - (instancetype)init {
 }
 
 - (void)addPlan:(NSObject<MDMPlan> *)plan toTarget:(id)target {
-  MDMTransactionLog *log = [MDMTransactionLog new];
-  log.plans = @[ [plan copy] ];
-  log.target = target;
-  [_logs addObject:log];
+  [self commonAddPlan:plan toTarget:target withName:nil];
+}
+
+- (void)addPlan:(NSObject<MDMPlan> *)plan named:(NSString *)name toTarget:(id)target {
+  NSParameterAssert(name.length > 0);
+  [self commonRemovePlan:plan fromTarget:target named:name];
```

Comment:
I think we can nuke this extra `removePlan` log here because the performer group should be responsible for removing previous named plans whenever a named plan is added.

Consider when we expose the `addPlan:named:` method on the scheduler: we can't expect clients to remember to call removePlan each time.